packages/ag-charts-community/src/chart/series/cartesian/areaSeries.ts

```diff
@@ -645,17 +645,17 @@ export class AreaSeries extends CartesianSeries<AreaSeriesNodeDataContext> {
         });
     }
 
-    getLegendData(): CategoryLegendDatum[] {
+    getLegendData(legendType: ChartLegendType): ChartLegendDatum[] {
         const { data, id, xKey, yKey, yName, marker, fill, stroke, fillOpacity, strokeOpacity, visible } = this;
 
-        if (!data?.length || !xKey || !yKey) {
+        if (!data?.length || !xKey || !yKey || legendType !== 'category') {
             return [];
         }
 
         // Area stacks should be listed in the legend in reverse order, for symmetry with the
         // vertical stack display order.
         return [
-            {
+            <CategoryLegendDatum>{
```

Comment:
Nit: Prefer `as CategoryLegendDatum` syntax (it's equivalent)?

Reply:
Alternatively, I believe there is some type-safety lost with any cast like this.

It might be better to follow this pattern, which ensures that the inline object conforms to the interface:
```
const legendDatun: CategoryLegendDatum = {
  legendType: 'category',
  id,
  // ...etc....
};
```